src/shogun/machine/GaussianProcessMachine.cpp

```diff
@@ -160,6 +165,8 @@ SGVector<float64_t> CGaussianProcessMachine::get_posterior_variances(
 
 	// get kernel matrix and create eigen representation of it
 	SGMatrix<float64_t> k_tsts=kernel->get_kernel_matrix();
+	//To DO in order to speed up the prediction process,
+	//we only need to obtain the diagonal elements of the kernel matrix
```

Comment:
@yorkerlin well, sorry I don't know how that algorithm works - if the working memory is the same as the matrix size, then I don't think adding a separate method for diagonal will gain something. But if you think you can manage with just an array of the size of the diagonal as the working memory (for computing the parameter gradient) - then it is useful.

Does this get_parameter_gradient work for *all* the kernel functions?